### ISSUE TITLE
`DateTimeHelper::toDateTime()` incorrectly parses year in 3.7.0+

### ISSUE THREAD

**Issue by mattstein** [craftcms/cms#9642]
### Description

A few people [have pointed out](https://github.com/craftcms/docs/issues/243) that [this Knowledge Base article’s yearly archive example](https://craftcms.com/knowledge-base/creating-an-archive-page-for-entries#yearly-archive-pages) no longer works in Craft 3.7:

```twig
<h1>{{ year }}</h1>

{% set entriesInYear = craft.entries()
    .section('blog')
    .after(year)
    .before(year+1)
    .all() %}

{% for month, entries in entriesInYear|group("postDate|date('F')") %}
    <h2>{{ month }}</h2>
    <ul>
        {% for entry in entries %}
            <li>{{ entry.getLink() }}</li>
        {% endfor %}
    </ul>
{% endfor %}
```

`{{ year }}` is a string like `'2021'`, and the example fails to return any entries with Craft 3.7 while it _does_ return entries as promised going back to the latest version of Craft 2. You can get around this in Craft 3.7 by including the day and month: `'2021-01-01'`, in which case entries will continue to be returned as expected.

It looks like it’s because [DateTimeHelper::toDateTime()](https://github.com/craftcms/cms/blob/main/src/helpers/DateTimeHelper.php#L85) behaves differently when parsing a string with only the year.

Quick test case:

```php
Craft::dd(DateTimeHelper::toDateTime('2021'));
```

**Craft 3.6.0**

```
DateTime#1
(
    [date] => '2020-12-31 16:00:00.000000'
    [timezone_type] => 3
    [timezone] => 'America/Los_Angeles'
)
```

**Craft 3.7.5**

```
DateTime#1
(
    [date] => '1969-12-31 16:33:41.000000'
    [timezone_type] => 3
    [timezone] => 'America/Los_Angeles'
)
```

It seems the problem was introduced specifically between Craft 3.7.0-beta.5 and Craft 3.7.0-beta.6.

### Additional info

- Craft version: 3.7.0-beta.6+
- PHP version: consistent with 7.2.34 and 8.0.3
- Database driver & version: MySQL 5.7, PostgreSQL 13

**Comment by angrybrad** [comment]
Guessing it's a regression from https://github.com/craftcms/cms/commit/8d20855904f09b89187754e142a8783f6ed343ee

**Comment by mattstein** [comment]
I suspected c57bfeb0bc77eb8db873ac02ae6c78e91b04bb20 but wasn’t sure.

**Comment by brandonkelly** [comment]
Yeah this was caused by c57bfeb0bc77eb8db873ac02ae6c78e91b04bb20. Just released 3.7.6 with a fix.
